WcaOnRails/app/views/registrations/_register_form.html.erb

```diff
@@ -12,7 +12,8 @@
   <%= f.input :comments, disabled: disabled %>
 
   <%# If the user has to pay fee here, this information will already be in the payment form %>
-  <% unless @registration.has_to_pay_fee_here? %>
+  <%# If there are no entry fees, no need to diplay this information %>
+  <% unless @registration.show_payment_form? || @registration.entry_fee == 0 %>
```

Comment:
@larspetrus may have already said some stuff that supercedes this, but it looks like you could use the `has_entry_fee?` method here instead of the `== 0` comparison. I'd probably rewrite it as:

```
if !registration.show_payment_form? && @registration.has_entry_fee?
```